pkg/operator/k8sutil/node.go

```diff
@@ -279,3 +280,74 @@ func NodeIsInRookNodeList(targetNodeName string, rookNodes []rookalpha.Node) boo
 	}
 	return false
 }
+
+// ValidateLabelKey validates Key of the Label against regex
+func ValidateLabelKey(s string) bool {
+	re := regexp.MustCompile(`^(((([a-zA-Z0-9]+[.]){1,}[a-zA-Z0-9]+){1}[\/]{1})?)([a-zA-Z0-9](([a-zA-Z0-9]*[\-_.][a-zA-Z0-9]+){1,})[a-zA-Z0-9]$)`)
+	return re.MatchString(s)
+}
+
+// ValidateLabelValue validates Value of the Label against regex
+func ValidateLabelValue(s string) bool {
+	re := regexp.MustCompile(`(?m)^[a-z0-9A-Z]+([a-z0-9A-Z]+[\-._])*[a-z0-9A-Z]+$`)
+	return re.MatchString(s)
+}
+
+// AddNodeAffinity will return v1.NodeAffinity or error
+func AddNodeAffinity(nodeAffinity string) (*v1.NodeAffinity, error) {
+	newNodeAffinity := &v1.NodeAffinity{
+		RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+			NodeSelectorTerms: []v1.NodeSelectorTerm{
+				{},
```

Comment:
Is there a need to have an empty `v1.NodeSelectorTerm` here (`{}`)?

If not, please remove this empty definition here.

Reply:
It is needed to prevent `IndexOutOfRange`. 
We need all match expressions to be in a single `NodeSelectorTerm` so that all labels are considered during scheduling.